subprojects/docs/src/docs/userguide/authoringMaintainableBuildScripts.adoc

```diff
@@ -157,6 +157,28 @@ Now that the build logic has been translated into a plugin, you can apply it in
 </sample>
 ++++
 
+[[sec:avoiding_use_of_gradlebuild]]
+=== Avoiding the use of `GradleBuild`
+
+A build script can define task for invocation other Gradle builds through the task type api:org.gradle.api.tasks.GradleBuild[]. The use of this type is generally discouraged. There are some corner cases where the invoked build doesn't expose the same runtime behavior as from the command line or through the Tooling API leading to unexpected results.
+
+Usually, there's a better way to model the requirement. The appropriate approach depends on the problem at hand. Here're some options:
+
+* Model the build as <<multi_project_builds,multi-project build>> if the intention is to execute tasks from different modules as unified build.
+* Use <<composite_builds,composite builds>> for projects that are physically separated but should occasionally be built as a single unit.
+
+[[sec:avoiding_inter_project_configuration]]
+=== Avoiding inter-project configuration
+
+Gradle does not restrict build script author to reach into the domain model from one project into another one in a <<multi_project_builds,multi-project build>>. Strongly-coupled projects hurts <<sec:parallel_execution,build execution performance>> as well as readability and maintainability of code.
```

Comment:
"restrict build script author to reach" -> "restrict build script authors from reaching" ?

Reply:
done